src/TLuaInterpreterMapper.cpp

```diff
@@ -3414,17 +3416,10 @@ int TLuaInterpreter::setGridMode(lua_State* L)
     } else {
         pA->gridMode = gridMode;
         pA->calcSpan();
-        if (host.mpMap->mpMapper) {
-            if (host.mpMap->mpMapper->mp2dMap) {
-                // Not needed IMHO - Slysven
-                //                host.mpMap->mpMapper->mp2dMap->init();
-                //                cout << "NEW GRID MAP: init" << endl;
-                // But this is:
-                host.mpMap->mpMapper->update();
-            }
-        }
+        host.mpMap->update();
     }
     host.mpMap->setUnsaved(__func__);
+    host.mpMap->update();
```

Comment:
You don't need to call this here - you've just done it in the `if` for the `else` case where the area exists - and if it doesn't the return-before-the-else will mean this point is not reached in that case!

Reply:
Updated.